### PR TITLE
Proposal for Section 6 - Signed Statement Issuance, Registration, and…

### PR DESCRIPTION
**Pull request by hannestschofenig**
Attempt to clean up Section 6

### CODE REVIEW

File: draft-ietf-scitt-architecture.md [hunk 1]
```diff
@@ -639,31 +639,24 @@ Although this weakens their guarantees against key revocation, or against a corr
...
 
-##  Envelope and Signed Statement Format
+##  Issuing Signed Statements
 
-The formats of Signed Statements and Receipts are based on CBOR Object Signing and Encryption (COSE {{-COSE}}).
-The choice of CBOR {{-CBOR}} is a trade-off between safety (in particular, non-malleability: each Signed Statement has a unique serialization), ease of processing and availability of implementations.
-
-At a high-level that is the context of this architecture, a Signed Statement is a COSE single-signed object (i.e., a `COSE_Sign1`) that contains the correct set of protected headers.
-Although Issuers and relying parties may attach unprotected headers to Signed Statements, Transparency Services and Verifiers MUST NOT rely on the presence or value of additional unprotected headers in Signed Statements during Registration and validation.
+Signed Statements are CBOR encoded {{-CBOR}} and protected by CBOR Object Signing and Encryption (COSE {{-COSE}}). Additionally, it contains a minimum amount of headers and Statements as its payload.
```
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:644]: Line 644 does not look right to me? Assuming that you want atleast one header and a statement as payload I will re-write the statement,

Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:644]:
```suggestion
Signed Statements are CBOR encoded {{-CBOR}} and protected by CBOR Object Signing and Encryption (COSE {{-COSE}}). Additionally, it contains at least one or more headers and a set of statements as its payload.
```

File: draft-ietf-scitt-architecture.md [hunk 2]
```diff
@@ -764,39 +706,33 @@ Once the Statement is serialized with the correct media-type/content-format, an
...
 The Transparency Service MUST store evidence of the DID resolution for the Issuer protected header of the Envelope and the resolved key manifest at the time of Registration for auditing.
 This MAY require that the service resolves the Issuer DID and record the resulting document, or rely on a cache of recent resolutions.
 
-3. Envelope signature verification, as described in COSE signature, using the signature algorithm and verification key of the Issuer DID document.
+2. Signed Statement signature verification, as described in COSE signature, using the signature algorithm and verification key of the Issuer DID document.
 
-4. Envelope validation.
-The service MUST check that the Envelope includes a Statement payload and the protected headers listed above.
-The service MAY additionally verify the Statement payload format and content.
+3. Signature validation.
```
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:720]:
```suggestion
3. Signed Statement validation.
```

File: draft-ietf-scitt-architecture.md (hunk 1 above)
Comment (hannestschofenig) [reply draft-ietf-scitt-architecture.md]: Accepted your wording suggestion

File: draft-ietf-scitt-architecture.md (hunk 2 above)
Comment (hannestschofenig) [reply draft-ietf-scitt-architecture.md]: Fixed this issue.

File: draft-ietf-scitt-architecture.md [hunk 3]
```diff
@@ -639,31 +639,24 @@ Although this weakens their guarantees against key revocation, or against a corr
 
 This section details the interoperability requirements for implementers of Signed Statements issuance and validation libraries, and of Transparency Services.
 
-##  Envelope and Signed Statement Format
+##  Issuing Signed Statements
 
-The formats of Signed Statements and Receipts are based on CBOR Object Signing and Encryption (COSE {{-COSE}}).
-The choice of CBOR {{-CBOR}} is a trade-off between safety (in particular, non-malleability: each Signed Statement has a unique serialization), ease of processing and availability of implementations.
```
Comment (umbellate) [inline draft-ietf-scitt-architecture.md:645]: I think missing closing parenthesis (')').

**Review by OR13 — APPROVED** [review]

File: draft-ietf-scitt-architecture.md [hunk 4]
```diff
@@ -639,31 +639,24 @@ Although this weakens their guarantees against key revocation, or against a corr
...
 
 All Signed Statements MUST include the following protected headers:
 
-- algorithm (label: `1`): Asymmetric signature algorithm used by the Issuer of a Signed Statement, as an integer, for example `-35` for ECDSA with SHA-384, see [COSE Algorithms Registry](#IANA.cose);
-- Issuer (label: `TBD`, temporary: `391`): DID (Decentralized Identifier {{DID-CORE}}) of the signer, as a string, for example `did:web:example.com`;
-- Feed (label: `TBD`, temporary: `392`): the Issuer's name for the Artifact, as a string;
-- payload type (label: `3`): media-type of Statement payload as a string, for example `application/spdx+json`
-- Registration Policy info (label: `TBD`, temporary: `393`): a map of additional attributes to help enforce Registration Policies;
+- algorithm (label: `1`): Asymmetric signature algorithm used by the Issuer of a Signed Statement, as an integer. For example, `-35` is the registered algorithm identifier for ECDSA with SHA-384, see [COSE Algorithms Registry](#IANA.cose).
+- Issuer (label: `TBD`, temporary: `391`): DID (Decentralized Identifier {{DID-CORE}}) of the signer, as a string. `did:web:example.com` is an example of a DID.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:650]: we should NOT do this... we should use https://datatracker.ietf.org/doc/draft-ietf-cose-cwt-claims-in-headers/

For now, I would recommend simply saying `kid` is mandatory and MUST be an absolute DID URL.

File: draft-ietf-scitt-architecture.md [hunk 5]
```diff
@@ -639,31 +639,24 @@ Although this weakens their guarantees against key revocation, or against a corr
...
+- Feed (label: `TBD`, temporary: `392`): The Issuer's name for the Artifact, as a string.
+- Payload type (label: `3`): Media type of payload, as a string. For example, `application/spdx+json` is the media type of SDPX in JSON encoding.
```
Comment (OR13) [inline draft-ietf-scitt-architecture.md:652]:
```suggestion
- content type (label: `3`): Media type of payload, as a string. For example, `application/spdx+json` is the media type of SDPX in JSON encoding.
```

Comment (OR13) [reply draft-ietf-scitt-architecture.md:652]: Do we want to allow COAP? see also https://datatracker.ietf.org/doc/draft-ietf-cose-typ-header-parameter/

`typ` is for the envelope, not the payload type, but the same CBOR media type stuff applies

Comment (OR13) [reply draft-ietf-scitt-architecture.md:652]: Tracking in https://github.com/ietf-wg-scitt/draft-ietf-scitt-architecture/issues/100

**Comment by hannestschofenig** [comment]
Orie, you realize that you are commenting on text that I didn't suggest. This text has already been there in the submitted draft

**Comment by OR13** [comment]
@hannestschofenig yes, I do realize that... The comment is still relevant to progressing the document, and its not a blocking change request.

File: draft-ietf-scitt-architecture.md [hunk 6]
```diff
@@ -639,31 +639,24 @@ Although this weakens their guarantees against key revocation, or against a corr
 
 This section details the interoperability requirements for implementers of Signed Statements issuance and validation libraries, and of Transparency Services.
 
-##  Envelope and Signed Statement Format
+##  Issuing Signed Statements
```
Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:642]:
```suggestion
##  Signed Statement Envelope
```

Comment (yogeshbdeshpande) [inline draft-ietf-scitt-architecture.md:642]: @hannestschofenig : As the Main heading for the section is `Signed Statement Issuance, Registration, and Verification`

I think, it makes sense to introduce this section as `Signed Statement Envelope` The reason been (a) We are not discussing Issuance per say here

(b) We do not have a fixed signed statement format, but we just have an Envelope which contains a `Payload`

**Review by henkbirkholz — APPROVED** [review]

**Review by SteveLasker — APPROVED** [review]
LGTM with resolution

**Review by yogeshbdeshpande — APPROVED** [review]
LGTM! After all the hard work !

**Pull request merged**